src/network/message.rs

```diff
@@ -42,21 +42,28 @@ pub const MAX_INV_SIZE: usize = 50_000;
 #[derive(PartialEq, Eq, Clone, Debug)]
 pub struct CommandString(Cow<'static, str>);
 
-impl fmt::Display for CommandString {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.write_str(self.0.as_ref())
-    }
-}
-
-impl From<&'static str> for CommandString {
-    fn from(f: &'static str) -> Self {
-        CommandString(f.into())
+impl CommandString {
+    /// Convert from various string types into a [CommandString].
+    ///
+    /// Supported types are:
+    /// - `&'static str`
+    /// - `String`
+    ///
+    /// Returns an empty error if and only if the string is
+    /// larger than 12 characters in length.
+    pub fn try_from<S: Into<Cow<'static, str>>>(s: S) -> Result<CommandString, ()> {
```

Comment:
Empty errors are bad (they do not implement `std::error::Error` trait since they have not Display), however, with rust 1.29 I think this can work for now)

Reply:
AFAIK s to implement unit structs (i.e. structs w/o any data) and implement Debug+Display for them (I do that with derive and amplify, but yep, for 1.29 you can't do a derive display with amplify)